packages/react-component-library/src/components/Timeline/context/reducer.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  addMonths,
+  getMonth,
+  setMonth,
+  startOfMonth,
+  endOfMonth,
+  differenceInWeeks,
+  addWeeks,
+  startOfWeek
+} from 'date-fns'
+
+import {
+  TimelineState,
+  TimelineAction,
+  TIMELINE_ACTIONS
+} from './types'
+
+export function getWeeks(
+  date: Date,
+  state: TimelineState
+) {
+  const { options: { range } } = state
+
+  const month = getMonth(date)
+
+  const startDate = startOfMonth(date)
+  const endDate = endOfMonth(setMonth(date, (month + range - 1)))
+
+  const diffInWeeks = differenceInWeeks(endDate, startDate)
+
+  const weeks = Array.from({ length: diffInWeeks + 1 })
+    .map((_, weekIndex) => {
+      const weekStart = startOfWeek(
+        addWeeks(startDate, weekIndex),
+        { weekStartsOn: 1 }
+      )
+
+      return {
+        weekIndex,
+        startDate: weekStart
+      }
+    })
+
+  return weeks
+}
+
+export function getMonths(
+  date: Date,
+  state: TimelineState
+) {
+  const { options: { range } } = state
+
+  const months = Array.from({ length: range })
+    .map((_, monthIndex) => {
+      const month = getMonth(date)
+      const newDate = setMonth(date, month + monthIndex)
+      const startDate = startOfMonth(newDate)
+
+      return {
+        monthIndex: month + monthIndex,
+        startDate
+      }
+    })
+
+  return {
+    startDate: months[0].startDate,
+    months
+  }
+}
+
+export function reducer(state: TimelineState, action: TimelineAction) {
+  switch (action.type) {
+    case TIMELINE_ACTIONS.SET_DATE:
```

Comment:
This `TIMELINE_ACTIONS` is covered but the other cases are not covered below.

Reply:
This relates to my earlier comment about spying on the reducer dispatch.